scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -31,18 +34,65 @@
                  "compute cluster, unable to get node from node name" node-name)
       false)))
 
+(defn num-waiting-synthetic-pods
+  "Given a compute cluster and a collection of pods, returns the number of
+  waiting synthetic pods in the cluster, or 0 if the cluster is not autoscaling"
+  [compute-cluster pods pool-name]
+  (if (cc/autoscaling? compute-cluster pool-name)
+    (let [synthetic-pods (filter controller/synthetic-pod->job-uuid pods)
+          waiting-synthetic-pods (filter (fn [^V1Pod pod]
+                                           (or (-> pod
+                                                   .getStatus
+                                                   nil?)
+                                               (-> pod
+                                                   api/pod->synthesized-pod-state
+                                                   :state
+                                                   (= :pod/waiting))))
+                                         synthetic-pods)
+          waiting-synthetic-pods-in-pool (filter (fn [^V1Pod pod]
+                                                   (let [pod-spec (.getSpec pod)
+                                                         tolerations (.getTolerations pod-spec)]
+                                                     (some (fn [^V1Toleration toleration]
+                                                             (and (= (.getKey toleration) "cook-pool")
+                                                                  (= (.getValue toleration) pool-name)))
+                                                           tolerations)))
+                                                 waiting-synthetic-pods)]
+      (count waiting-synthetic-pods-in-pool))
```

Comment:
Nitpick: I like naming the anonymous functions so that it is easy to locate during debugging.

Reply:
Will do

Reply:
Done